gapic/utils/code.py

```diff
@@ -21,3 +23,29 @@ def empty(content: str) -> bool:
     """
     return not any([i.lstrip() and not i.lstrip().startswith('#')
                     for i in content.split('\n')])
+
+
+T = TypeVar('T')
+
+
+def partition(predicate: Callable[[T], bool],
+              iterator: Iterable[T]) -> Tuple[List[T], List[T]]:
+    """Partitions an iterable into two lists based on a predicate
+
+    Args:
+        predicate Callable((T), bool) : A callable predicate on a single argument
```

Comment:
nit: In other methods you have the type inside parentheses. Do so here too for consistency.

(BTW, is the style to repeat the type even though it's already listed in the type hints in the function signature?)

Reply:
A quick experiment indicates that pydoc is aware of type annotations. I'll remove the type info from the docstrings.

Reply:
Python type hints are new-ish and not all tooling supports them yet, so repetition is semi common.

These things are parsed and have meaning though, and the one here is wrong; it should be `Callable[[T], bool]`.